### ISSUE TITLE
Support `healthcheck.timeout` of `aws_lb_target_group` when `protocol` is `TCP`

### ISSUE THREAD

**Issue by posquit0** [hashicorp/terraform-provider-aws#23222]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)

* aws_lb_target_group

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
```

### Expected Behavior
<img width="1129" alt="image" src="https://user-images.githubusercontent.com/1484002/154220487-5e437d5c-f308-4bb2-aadf-26eb2f8084ce.png">
<img width="655" alt="image" src="https://user-images.githubusercontent.com/1484002/154221042-862cf0e7-160d-48f2-83c8-047e51f1ce5a.png">

I can confirm that AWS support `healthcheck.timeout` for target group of gateway loadbalancer.

### Actual Behavior

I created `aws_lb_target_group` for gateway load balancer.

```
resource "aws_lb_target_group" "this" {
  name = var.name

  vpc_id = var.vpc_id

  target_type = "ip"
  protocol    = "GENEVE"
  port        = local.port

  ## Attributes
  deregistration_delay = var.deregistration_delay

  health_check {
    enabled = true

    port     = try(var.health_check.port, local.port)
    protocol = try(var.health_check.protocol, "TCP")

    healthy_threshold   = try(var.health_check.healthy_threshold, 3)
    unhealthy_threshold = try(var.health_check.unhealthy_threshold, 3)
    interval            = try(var.health_check.interval, 30)
    timeout             = try(var.health_check.timeout, 10)
  }
}
```

It occurs following an error.
```
│ Error: : health_check.timeout is not supported for target_groups with TCP protocol
│
│   with module.gwlb_ip.aws_lb_target_group.this,
│   on ../../../../../xxxxxxxx/modules/gwlb-ip-target-group/main.tf line 50, in resource "aws_lb_target_group" "this":
│   50: resource "aws_lb_target_group" "this" {
```

**Comment by posquit0** [comment]
<img width="1128" alt="image" src="https://user-images.githubusercontent.com/1484002/154222208-9d9d7243-be9a-4546-9589-34f29e4d1591.png">

One more case:
- `healthy_threshold` and `unhealthy_threshold` do not need to be the same. However, it occurs following error.

```
│ Error: arn:aws:elasticloadbalancing:ap-yyyyyy:xxxxxxxx:targetgroup/claud-tg-gwlb-ip/0068703ecc02db75f7: health_check.healthy_threshold 4 and health_check.unhealthy_threshold 5 must be the same for target_groups with TCP protocol
│
│   with module.gwlb_ip.aws_lb_target_group.this,
│   on ../../../../../txxxxxxxx/modules/gwlb-ip-target-group/main.tf line 50, in resource "aws_lb_target_group" "this":
│   50: resource "aws_lb_target_group" "this" {
│

```

**Comment by posquit0** [comment]
`healthcheck.interval` doesn't apply to the actual target group. The output from terraform is different from the AWS web console.

<img width="1134" alt="image" src="https://user-images.githubusercontent.com/1484002/154222976-de217652-cfa6-4442-aaa6-f5e3ad6c34bb.png">

```
    {
      "enabled" = true
      "healthy_threshold" = 4
      "interval" = 20
      "matcher" = ""
      "path" = ""
      "port" = "81"
      "protocol" = "TCP"
      "timeout" = 5
      "unhealthy_threshold" = 4
    },
```
```

**Comment by nikhilo** [comment]
I noticed that if the target group already exists, then terraform plan gives such error. If you delete the target group, (and also delete any listener that is using the target group) and try to create it using the timeout parameter it should be able to create it.
This is merely a workaround. There's definitely something to fix in the code.

**Comment by johnsonaj** [comment]
Issue resolved as part of #28018. Released in provider version `v4.46.0`
